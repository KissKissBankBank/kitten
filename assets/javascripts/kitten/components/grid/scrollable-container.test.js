import React from 'react'
import renderer from 'react-test-renderer'
import { ScrollableContainer }
  from 'kitten/components/grid/scrollable-container'

describe('<ScrollableContainer />', () => {
  let component

  describe('by default', () => {
    beforeEach(() => {
      component = renderer.create(
        <ScrollableContainer />
      ).toJSON()
    })

    it('matches with snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })

  describe('with some props', () => {
    beforeEach(() => {
      component = renderer.create(
        <ScrollableContainer
          className="custom__class"
          style={{ color: 'red' }}
          aria-hidden
        />
      ).toJSON()
    })

    it('matches with snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })

  describe('with children', () => {
    beforeEach(() => {
      component = renderer.create(
        <ScrollableContainer>
          <nav>
            <ul>
              <li>Item 1</li>
              <li>Item 2</li>
              <li>Item 3</li>
            </ul>
          </nav>
        </ScrollableContainer>
      ).toJSON()
    })

    it('matches with snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })
})
