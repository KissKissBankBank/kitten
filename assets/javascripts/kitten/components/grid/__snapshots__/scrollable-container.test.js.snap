// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ScrollableContainer /> by default matches with snapshot 1`] = `
<div
  className="k-Container k-Container--no-padding"
  style={
    Object {
      "position": "relative",
    }
  }
>
  <div
    className="k-ScrollableContainer"
    style={
      Object {
        "MsOverflowStyle": "none",
        "WebkitOverflowScrolling": "touch",
        "display": "flex",
        "overflowX": "auto",
        "whiteSpace": "nowrap",
      }
    }
  />
  <div
    style={
      Object {
        "background": "linear-gradient(90deg, #fff, rgba(255, 255, 255, 0))",
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "top": 0,
        "width": 20,
      }
    }
  />
  <div
    style={
      Object {
        "background": "linear-gradient(90deg, rgba(255, 255, 255, 0), #fff)",
        "bottom": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 20,
      }
    }
  />
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": ".k-ScrollableContainer::-webkit-scrollbar{display: none;}",
      }
    }
  />
</div>
`;

exports[`<ScrollableContainer /> with children matches with snapshot 1`] = `
<div
  className="k-Container k-Container--no-padding"
  style={
    Object {
      "position": "relative",
    }
  }
>
  <div
    className="k-ScrollableContainer"
    style={
      Object {
        "MsOverflowStyle": "none",
        "WebkitOverflowScrolling": "touch",
        "display": "flex",
        "overflowX": "auto",
        "whiteSpace": "nowrap",
      }
    }
  >
    <nav>
      <ul>
        <li>
          Item 1
        </li>
        <li>
          Item 2
        </li>
        <li>
          Item 3
        </li>
      </ul>
    </nav>
  </div>
  <div
    style={
      Object {
        "background": "linear-gradient(90deg, #fff, rgba(255, 255, 255, 0))",
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "top": 0,
        "width": 20,
      }
    }
  />
  <div
    style={
      Object {
        "background": "linear-gradient(90deg, rgba(255, 255, 255, 0), #fff)",
        "bottom": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 20,
      }
    }
  />
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": ".k-ScrollableContainer::-webkit-scrollbar{display: none;}",
      }
    }
  />
</div>
`;

exports[`<ScrollableContainer /> with some props matches with snapshot 1`] = `
<div
  aria-hidden={true}
  className="k-Container k-Container--no-padding custom__class"
  style={
    Object {
      "color": "red",
      "position": "relative",
    }
  }
>
  <div
    className="k-ScrollableContainer"
    style={
      Object {
        "MsOverflowStyle": "none",
        "WebkitOverflowScrolling": "touch",
        "display": "flex",
        "overflowX": "auto",
        "whiteSpace": "nowrap",
      }
    }
  />
  <div
    style={
      Object {
        "background": "linear-gradient(90deg, #fff, rgba(255, 255, 255, 0))",
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "top": 0,
        "width": 20,
      }
    }
  />
  <div
    style={
      Object {
        "background": "linear-gradient(90deg, rgba(255, 255, 255, 0), #fff)",
        "bottom": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 20,
      }
    }
  />
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": ".k-ScrollableContainer::-webkit-scrollbar{display: none;}",
      }
    }
  />
</div>
`;
